### PR TITLE
[monotouch-test] Check for Metal compatibility for all platforms before creating indirect command buffers. Fixes #xamarin/maccore@2235.

### PR DESCRIPTION
**Pull request by rolfbjarne**
This fixes an iOS assertion (crash) when running on iPhone 6 with iOS 12+:

    2020-06-15 13:52:35.138 monotouchtest[6887:5713672] failed assertion 0 at line 54 in NopIndirectCommandBuffer

    =================================================================
    	Native Crash Reporting
    =================================================================
    Got a abrt while executing native code. This usually indicates
    a fatal error in the mono runtime or one of the native libraries
    used by your application.
    =================================================================

    =================================================================
    	Native stacktrace:
    =================================================================
    	0x1060b23fc - /private/var/containers/Bundle/Application/B3C2F137-4C0F-48EF-A50A-8972FF268069/monotouchtest.app/libmonosgen-2.0.dylib : mono_dump_native_crash_info
    	0x1060a8674 - /private/var/containers/Bundle/Application/B3C2F137-4C0F-48EF-A50A-8972FF268069/monotouchtest.app/libmonosgen-2.0.dylib : mono_handle_native_crash
    	0x1060b1948 - /private/var/containers/Bundle/Application/B3C2F137-4C0F-48EF-A50A-8972FF268069/monotouchtest.app/libmonosgen-2.0.dylib : sigabrt_signal_handler
    	0x20049e9fc - /usr/lib/system/libsystem_platform.dylib : <redacted>
    	0x2004a4094 - /usr/lib/system/libsystem_pthread.dylib : <redacted>
    	0x200383ea8 - /usr/lib/system/libsystem_c.dylib : abort
    	0x2029c5dac - /System/Library/Frameworks/Metal.framework/Metal : MTLGetWarningMode
    	0x2210cff78 - /System/Library/Extensions/AGXMetalA8.bundle/AGXMetalA8 :
    	0x20299fb80 - /System/Library/Frameworks/Metal.framework/Metal : <redacted>
    	0x103143c58 - /private/var/containers/Bundle/Application/B3C2F137-4C0F-48EF-A50A-8972FF268069/monotouchtest.app/libXamarin.iOS.dll.dylib : wrapper_managed_to_native_ObjCRuntime_Messaging_IntPtr_objc_msgSend_IntPtr_nuint_UInt64_intptr_intptr_intptr_System_nuint_ulong
    	0x102f4cbdc - /private/var/containers/Bundle/Application/B3C2F137-4C0F-48EF-A50A-8972FF268069/monotouchtest.app/libXamarin.iOS.dll.dylib : Metal_MTLDevice_Extensions_CreateIndirectCommandBuffer_Metal_IMTLDevice_Metal_MTLIndirectCommandBufferDescriptor_System_nuint_Metal_MTLResourceOptions
    	0x101a8bd58 - /private/var/containers/Bundle/Application/B3C2F137-4C0F-48EF-A50A-8972FF268069/monotouchtest.app/libmonotouchtest.exe.dylib : MonoTouchFixtures_Metal_MTLDeviceTests_ReturnReleaseTest
    	[...]

Fixes https://github.com/xamarin/maccore/issues/2235.

### CODE REVIEW

File: tests/monotouch-test/Metal/MTLDeviceTests.cs [hunk 1]
```diff
@@ -272,9 +272,7 @@ public void ReturnReleaseTest ()
 			}
 
 			if (TestRuntime.CheckXcodeVersion (10, 0)) {
-#if __MACOS__
-				if (device.SupportsFeatureSet (MTLFeatureSet.macOS_GPUFamily2_v1))
-#endif
+				if (device.SupportsFamily (MTLGpuFamily.Common2))
```
Comment (spouliot) [inline tests/monotouch-test/Metal/MTLDeviceTests.cs:275]: minor: indentation
maybe move it with the `TestRuntime` check earlier (so no other lines needs to be changed)

**Comment by xamarin-release-manager** [comment]
❌ [Build was (probably) aborted](https://jenkins.internalx.com/job/macios/job/PR-8863/2/display/redirect)

🔥 [Jenkins job](https://jenkins.internalx.com/job/macios/job/PR-8863/2/display/redirect) (on internal Jenkins) failed in stage(s) 'Running XM tests on '10.9'' 🔥 : hudson.AbortException: Xamarin.Mac tests on macOS 10.9 failed (xammac_tests)

✅ [Provisioning succeeded](https://jenkins.internalx.com/job/macios/job/PR-8863/2//console)
✅ [Build succeeded](https://jenkins.internalx.com/job/macios/job/PR-8863/2//console)
✅ Packages built successfully
<details><summary>View packages</summary>

* [xamarin.ios-13.21.1.136.pkg (Not notarized)](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-8863/9fd1910b0b45ee5c1894b0239f88039a08794d76/2/package/xamarin.ios-13.21.1.136.pkg)
* [xamarin.mac-6.21.1.136.pkg (Not notarized)](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-8863/9fd1910b0b45ee5c1894b0239f88039a08794d76/2/package/xamarin.mac-6.21.1.136.pkg)
* [Microsoft.iOS.Ref.13.5.100-ci.pr.gh8863.136+sha.9fd1910b0.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-8863/9fd1910b0b45ee5c1894b0239f88039a08794d76/2/package/Microsoft.iOS.Ref.13.5.100-ci.pr.gh8863.136+sha.9fd1910b0.nupkg)
* [Microsoft.iOS.Runtime.ios-arm.13.5.100-ci.pr.gh8863.136+sha.9fd1910b0.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-8863/9fd1910b0b45ee5c1894b0239f88039a08794d76/2/package/Microsoft.iOS.Runtime.ios-arm.13.5.100-ci.pr.gh8863.136+sha.9fd1910b0.nupkg)
* [Microsoft.iOS.Runtime.ios-arm64.13.5.100-ci.pr.gh8863.136+sha.9fd1910b0.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-8863/9fd1910b0b45ee5c1894b0239f88039a08794d76/2/package/Microsoft.iOS.Runtime.ios-arm64.13.5.100-ci.pr.gh8863.136+sha.9fd1910b0.nupkg)
* [Microsoft.iOS.Runtime.ios-x64.13.5.100-ci.pr.gh8863.136+sha.9fd1910b0.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-8863/9fd1910b0b45ee5c1894b0239f88039a08794d76/2/package/Microsoft.iOS.Runtime.ios-x64.13.5.100-ci.pr.gh8863.136+sha.9fd1910b0.nupkg)
* [Microsoft.iOS.Runtime.ios-x86.13.5.100-ci.pr.gh8863.136+sha.9fd1910b0.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-8863/9fd1910b0b45ee5c1894b0239f88039a08794d76/2/package/Microsoft.iOS.Runtime.ios-x86.13.5.100-ci.pr.gh8863.136+sha.9fd1910b0.nupkg)
* [Microsoft.iOS.Sdk.13.5.100-ci.pr.gh8863.136+sha.9fd1910b0.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-8863/9fd1910b0b45ee5c1894b0239f88039a08794d76/2/package/Microsoft.iOS.Sdk.13.5.100-ci.pr.gh8863.136+sha.9fd1910b0.nupkg)
* [Microsoft.macOS.Ref.10.15.100-ci.pr.gh8863.175+sha.9fd1910b0.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-8863/9fd1910b0b45ee5c1894b0239f88039a08794d76/2/package/Microsoft.macOS.Ref.10.15.100-ci.pr.gh8863.175+sha.9fd1910b0.nupkg)
* [Microsoft.macOS.Runtime.osx-x64.10.15.100-ci.pr.gh8863.175+sha.9fd1910b0.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-8863/9fd1910b0b45ee5c1894b0239f88039a08794d76/2/package/Microsoft.macOS.Runtime.osx-x64.10.15.100-ci.pr.gh8863.175+sha.9fd1910b0.nupkg)
* [Microsoft.macOS.Sdk.10.15.100-ci.pr.gh8863.175+sha.9fd1910b0.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-8863/9fd1910b0b45ee5c1894b0239f88039a08794d76/2/package/Microsoft.macOS.Sdk.10.15.100-ci.pr.gh8863.175+sha.9fd1910b0.nupkg)
* [Microsoft.tvOS.Ref.13.4.100-ci.pr.gh8863.175+sha.9fd1910b0.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-8863/9fd1910b0b45ee5c1894b0239f88039a08794d76/2/package/Microsoft.tvOS.Ref.13.4.100-ci.pr.gh8863.175+sha.9fd1910b0.nupkg)
* [Microsoft.tvOS.Runtime.tvos-arm64.13.4.100-ci.pr.gh8863.175+sha.9fd1910b0.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-8863/9fd1910b0b45ee5c1894b0239f88039a08794d76/2/package/Microsoft.tvOS.Runtime.tvos-arm64.13.4.100-ci.pr.gh8863.175+sha.9fd1910b0.nupkg)
* [Microsoft.tvOS.Runtime.tvos-x64.13.4.100-ci.pr.gh8863.175+sha.9fd1910b0.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-8863/9fd1910b0b45ee5c1894b0239f88039a08794d76/2/package/Microsoft.tvOS.Runtime.tvos-x64.13.4.100-ci.pr.gh8863.175+sha.9fd1910b0.nupkg)
* [Microsoft.tvOS.Sdk.13.4.100-ci.pr.gh8863.175+sha.9fd1910b0.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-8863/9fd1910b0b45ee5c1894b0239f88039a08794d76/2/package/Microsoft.tvOS.Sdk.13.4.100-ci.pr.gh8863.175+sha.9fd1910b0.nupkg)
* [Microsoft.watchOS.Ref.6.2.100-ci.pr.gh8863.175+sha.9fd1910b0.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-8863/9fd1910b0b45ee5c1894b0239f88039a08794d76/2/package/Microsoft.watchOS.Ref.6.2.100-ci.pr.gh8863.175+sha.9fd1910b0.nupkg)
* [Microsoft.watchOS.Runtime.watchos-arm.6.2.100-ci.pr.gh8863.175+sha.9fd1910b0.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-8863/9fd1910b0b45ee5c1894b0239f88039a08794d76/2/package/Microsoft.watchOS.Runtime.watchos-arm.6.2.100-ci.pr.gh8863.175+sha.9fd1910b0.nupkg)
* [Microsoft.watchOS.Runtime.watchos-x86.6.2.100-ci.pr.gh8863.175+sha.9fd1910b0.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-8863/9fd1910b0b45ee5c1894b0239f88039a08794d76/2/package/Microsoft.watchOS.Runtime.watchos-x86.6.2.100-ci.pr.gh8863.175+sha.9fd1910b0.nupkg)
* [Microsoft.watchOS.Sdk.6.2.100-ci.pr.gh8863.175+sha.9fd1910b0.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-8863/9fd1910b0b45ee5c1894b0239f88039a08794d76/2/package/Microsoft.watchOS.Sdk.6.2.100-ci.pr.gh8863.175+sha.9fd1910b0.nupkg)

</details>

✅ [API Diff (from stable)](http://xamarin-storage/jenkins/xamarin-macios/PR-8863/9fd1910b0b45ee5c1894b0239f88039a08794d76/2/jenkins-results/api-diff/index.html)
✅ [API Diff (from PR only)](http://xamarin-storage/jenkins/xamarin-macios/PR-8863/9fd1910b0b45ee5c1894b0239f88039a08794d76/2/jenkins-results/apicomparison/api-diff.html) (no change)
✅ [Generator Diff](http://xamarin-storage/jenkins/xamarin-macios/PR-8863/9fd1910b0b45ee5c1894b0239f88039a08794d76/2/jenkins-results/generator-diff/index.html) (no change)
🔥 [Xamarin.Mac tests on 10.9 failed: Xamarin.Mac tests on macOS 10.9 failed (xammac_tests)](https://jenkins.internalx.com/job/macios/job/PR-8863/2/display/redirect) 🔥
✅ [Test run succeeded](http://xamarin-storage/jenkins/xamarin-macios/PR-8863/9fd1910b0b45ee5c1894b0239f88039a08794d76/2/jenkins-results/tests/index.html)

**Comment by xamarin-release-manager** [comment]
❌ [Build was (probably) aborted](https://jenkins.internalx.com/job/macios/job/PR-8863/3/display/redirect)

🔥 [Jenkins job](https://jenkins.internalx.com/job/macios/job/PR-8863/3/display/redirect) (on internal Jenkins) failed in stage(s) 'Publish Nugets, Running XM tests on '10.9'' 🔥

✅ [Provisioning succeeded](https://jenkins.internalx.com/job/macios/job/PR-8863/3//console)
✅ [Build succeeded](https://jenkins.internalx.com/job/macios/job/PR-8863/3//console)
✅ Packages built successfully
<details><summary>View packages</summary>

* [xamarin.ios-13.21.1.136.pkg (Not notarized)](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-8863/9fd1910b0b45ee5c1894b0239f88039a08794d76/3/package/xamarin.ios-13.21.1.136.pkg)
* [xamarin.mac-6.21.1.136.pkg (Not notarized)](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-8863/9fd1910b0b45ee5c1894b0239f88039a08794d76/3/package/xamarin.mac-6.21.1.136.pkg)
* [Microsoft.iOS.Ref.13.5.100-ci.pr.gh8863.136+sha.9fd1910b0.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-8863/9fd1910b0b45ee5c1894b0239f88039a08794d76/3/package/Microsoft.iOS.Ref.13.5.100-ci.pr.gh8863.136+sha.9fd1910b0.nupkg)
* [Microsoft.iOS.Runtime.ios-arm.13.5.100-ci.pr.gh8863.136+sha.9fd1910b0.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-8863/9fd1910b0b45ee5c1894b0239f88039a08794d76/3/package/Microsoft.iOS.Runtime.ios-arm.13.5.100-ci.pr.gh8863.136+sha.9fd1910b0.nupkg)
* [Microsoft.iOS.Runtime.ios-arm64.13.5.100-ci.pr.gh8863.136+sha.9fd1910b0.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-8863/9fd1910b0b45ee5c1894b0239f88039a08794d76/3/package/Microsoft.iOS.Runtime.ios-arm64.13.5.100-ci.pr.gh8863.136+sha.9fd1910b0.nupkg)
* [Microsoft.iOS.Runtime.ios-x64.13.5.100-ci.pr.gh8863.136+sha.9fd1910b0.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-8863/9fd1910b0b45ee5c1894b0239f88039a08794d76/3/package/Microsoft.iOS.Runtime.ios-x64.13.5.100-ci.pr.gh8863.136+sha.9fd1910b0.nupkg)
* [Microsoft.iOS.Runtime.ios-x86.13.5.100-ci.pr.gh8863.136+sha.9fd1910b0.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-8863/9fd1910b0b45ee5c1894b0239f88039a08794d76/3/package/Microsoft.iOS.Runtime.ios-x86.13.5.100-ci.pr.gh8863.136+sha.9fd1910b0.nupkg)
* [Microsoft.iOS.Sdk.13.5.100-ci.pr.gh8863.136+sha.9fd1910b0.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-8863/9fd1910b0b45ee5c1894b0239f88039a08794d76/3/package/Microsoft.iOS.Sdk.13.5.100-ci.pr.gh8863.136+sha.9fd1910b0.nupkg)
* [Microsoft.macOS.Ref.10.15.100-ci.pr.gh8863.175+sha.9fd1910b0.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-8863/9fd1910b0b45ee5c1894b0239f88039a08794d76/3/package/Microsoft.macOS.Ref.10.15.100-ci.pr.gh8863.175+sha.9fd1910b0.nupkg)
* [Microsoft.macOS.Runtime.osx-x64.10.15.100-ci.pr.gh8863.175+sha.9fd1910b0.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-8863/9fd1910b0b45ee5c1894b0239f88039a08794d76/3/package/Microsoft.macOS.Runtime.osx-x64.10.15.100-ci.pr.gh8863.175+sha.9fd1910b0.nupkg)
* [Microsoft.macOS.Sdk.10.15.100-ci.pr.gh8863.175+sha.9fd1910b0.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-8863/9fd1910b0b45ee5c1894b0239f88039a08794d76/3/package/Microsoft.macOS.Sdk.10.15.100-ci.pr.gh8863.175+sha.9fd1910b0.nupkg)
* [Microsoft.tvOS.Ref.13.4.100-ci.pr.gh8863.175+sha.9fd1910b0.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-8863/9fd1910b0b45ee5c1894b0239f88039a08794d76/3/package/Microsoft.tvOS.Ref.13.4.100-ci.pr.gh8863.175+sha.9fd1910b0.nupkg)
* [Microsoft.tvOS.Runtime.tvos-arm64.13.4.100-ci.pr.gh8863.175+sha.9fd1910b0.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-8863/9fd1910b0b45ee5c1894b0239f88039a08794d76/3/package/Microsoft.tvOS.Runtime.tvos-arm64.13.4.100-ci.pr.gh8863.175+sha.9fd1910b0.nupkg)
* [Microsoft.tvOS.Runtime.tvos-x64.13.4.100-ci.pr.gh8863.175+sha.9fd1910b0.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-8863/9fd1910b0b45ee5c1894b0239f88039a08794d76/3/package/Microsoft.tvOS.Runtime.tvos-x64.13.4.100-ci.pr.gh8863.175+sha.9fd1910b0.nupkg)
* [Microsoft.tvOS.Sdk.13.4.100-ci.pr.gh8863.175+sha.9fd1910b0.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-8863/9fd1910b0b45ee5c1894b0239f88039a08794d76/3/package/Microsoft.tvOS.Sdk.13.4.100-ci.pr.gh8863.175+sha.9fd1910b0.nupkg)
* [Microsoft.watchOS.Ref.6.2.100-ci.pr.gh8863.175+sha.9fd1910b0.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-8863/9fd1910b0b45ee5c1894b0239f88039a08794d76/3/package/Microsoft.watchOS.Ref.6.2.100-ci.pr.gh8863.175+sha.9fd1910b0.nupkg)
* [Microsoft.watchOS.Runtime.watchos-arm.6.2.100-ci.pr.gh8863.175+sha.9fd1910b0.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-8863/9fd1910b0b45ee5c1894b0239f88039a08794d76/3/package/Microsoft.watchOS.Runtime.watchos-arm.6.2.100-ci.pr.gh8863.175+sha.9fd1910b0.nupkg)
* [Microsoft.watchOS.Runtime.watchos-x86.6.2.100-ci.pr.gh8863.175+sha.9fd1910b0.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-8863/9fd1910b0b45ee5c1894b0239f88039a08794d76/3/package/Microsoft.watchOS.Runtime.watchos-x86.6.2.100-ci.pr.gh8863.175+sha.9fd1910b0.nupkg)
* [Microsoft.watchOS.Sdk.6.2.100-ci.pr.gh8863.175+sha.9fd1910b0.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-8863/9fd1910b0b45ee5c1894b0239f88039a08794d76/3/package/Microsoft.watchOS.Sdk.6.2.100-ci.pr.gh8863.175+sha.9fd1910b0.nupkg)

</details>

🔥 [Failed to publish to nuget](https://jenkins.internalx.com/job/macios/job/PR-8863/3//console) 🔥
⚠️ [Failed to publish NuGet packages. Please review log for details.](https://jenkins.internalx.com/job/macios/job/PR-8863/3/display/redirect) 🔥
✅ [API Diff (from stable)](http://xamarin-storage/jenkins/xamarin-macios/PR-8863/9fd1910b0b45ee5c1894b0239f88039a08794d76/3/jenkins-results/api-diff/index.html)
✅ [API Diff (from PR only)](http://xamarin-storage/jenkins/xamarin-macios/PR-8863/9fd1910b0b45ee5c1894b0239f88039a08794d76/3/jenkins-results/apicomparison/api-diff.html) (no change)
✅ [Generator Diff](http://xamarin-storage/jenkins/xamarin-macios/PR-8863/9fd1910b0b45ee5c1894b0239f88039a08794d76/3/jenkins-results/generator-diff/index.html) (no change)
🔥 [Xamarin.Mac tests on 10.9 failed: Xamarin.Mac tests on macOS 10.9 failed (xammac_tests)](https://jenkins.internalx.com/job/macios/job/PR-8863/3/display/redirect) 🔥
🔥 [Test run failed](http://xamarin-storage/jenkins/xamarin-macios/PR-8863/9fd1910b0b45ee5c1894b0239f88039a08794d76/3/jenkins-results/tests/index.html) 🔥

# Test results

<details>
<summary>2 tests failed, 182 tests passed.</summary>

## Failed tests

 * monotouch-test/iOS Unified 32-bits - simulator/Debug (static registrar): TimedOut
 * monotouch-test/iOS Unified 32-bits - simulator/Release (all optimizations): Failed
</details>

File: tests/monotouch-test/Metal/MTLDeviceTests.cs (hunk 1 above)
Comment (rolfbjarne) [reply tests/monotouch-test/Metal/MTLDeviceTests.cs]: Fixed.

**Comment by rolfbjarne** [comment]
Test failures are unrelated to this PR.

* monotouch-test/iOS 32-bits/Debug: https://github.com/xamarin/maccore/issues/2219
* monotouch-test/iOS 32-bits/Release: https://github.com/xamarin/maccore/issues/2160
* macOS 10.9: https://github.com/xamarin/maccore/issues/2221

**Pull request merged**